src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,50 +1,96 @@
-import { useState } from 'react';
+import React, { FormEvent, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type NewMovieProps = {
+  onAdd: (movie:Movie) => void
+};
+
+export const NewMovie: React.FC<NewMovieProps> = ({
+  onAdd,
+}) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const isValid = title && imgUrl && imdbUrl && imdbId;
+
+  const resetForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+    setCount(current => current + 1);
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+    const movie: Movie = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    resetForm();
+    onAdd(movie);
+  };
```

Comment:
Do you need this count?
I think - it's extra
```suggestion
  const [title, setTitle] = useState('');
  const [description, setDescription] = useState('');
  const [imgUrl, setImgUrl] = useState('');
  const [imdbUrl, setImdbUrl] = useState('');
  const [imdbId, setImdbId] = useState('');
  const isValid = title && imgUrl && imdbUrl && imdbId;

  const resetForm = () => {
    setTitle('');
    setDescription('');
    setImgUrl('');
    setImdbUrl('');
    setImdbId('');
  };

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    
    const movie: Movie = {
      title,
      description,
      imgUrl,
      imdbUrl,
      imdbId,
    };

    resetForm();
    onAdd(movie);
  };
```